server/api/src/rsDataObjOpen.cpp

```diff
@@ -577,21 +690,51 @@ int _rsDataObjOpen(
                     unlock_data_obj();
                 }
             }
-            return l1descInx;
-        }
+            
+            // TODO Shouldn't the l1descInx be compared to three instead of zero
+            // given that the first three file descriptors are special?
+            if (l1descInx >= 0) {
```

Comment:
See comment.

Reply:
this will likely be an istream test.

Reply:
This is likely something that will need to be addressed throughout the code base. Also, this is not necessarily wrong.